nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/NiFiPropertiesLoader.java

```diff
@@ -158,9 +158,7 @@ private static String getDefaultProviderKey() {
     }
 
     private void initializeSensitivePropertyProviderFactory() {
-        if (sensitivePropertyProviderFactory == null) {
-            sensitivePropertyProviderFactory = new AESSensitivePropertyProviderFactory(keyHex);
-        }
+        sensitivePropertyProviderFactory = new AESSensitivePropertyProviderFactory(keyHex);
```

Comment:
It will if two competing resources are attempting to access it, but this scenario should not occur (there should not be two resources with different keys trying to load the data in the real application, as there is only one master key). The reason it was failing in your original test was due to sequential access not being cleared after migration. If we are concerned about race conditions, this should not be `static` at all and more of the organization needs to change.